comp/forwarder/defaultforwarder/forwarder_health.go

```diff
@@ -289,7 +289,17 @@ func (fh *forwarderHealth) checkValidAPIKey() bool {
 	fh.keyMapMutex.Unlock()
 
 	for domain, apiKeys := range keysPerDomain {
-		apiError, validKey = fh.checkValidAPIKeys(domain, apiKeys)
+		endpointApiError, endpointValidKey := fh.checkValidAPIKeys(domain, apiKeys)
+
+		// Only set the valid if the endpoint valid is true to ensure
+		// we don't unset the flag if one endpoint is not valid.
+		if endpointValidKey {
+			validKey = true
```

Comment:
An alternative would be:

```
validKey = validKey || endpointValidKey
```

same for the error below.

Discard if you prefer your implementation.